File: cp/Round#863 (Div. 3)/insertDigit.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;
 
int main() {
    int testCases; cin >> testCases;
    while(testCases--){
        int len; cin >> len;
        string addNumber; cin >> addNumber;
        string ogNumber; cin >> ogNumber;
        string equisde;
        int flag = len; //quizas tiene que ser len+1
        int flagFirst = 0;
        vector<string> result;

        for(int i = 0; i < len; i++){
            if(ogNumber[i] <= addNumber[0]){
                flag = i;
            }
            equisde = ogNumber[i];
            result.push_back(equisde);
        }

        result.insert(result.begin() + flag + addNumber);        
        
        for (int i = 0; i < len+1; i++){
            cout<<result[i];
        }
    }
}

// 1 <- meter

// 828169 <- numero

// 8281691 <-bueno
// 8281169


// 8281271277321 <- bueno
// 8281127127732

// 27240273

// 272402731
// 272410273

File: cp/Round#863 (Div. 3)/insertDigit.py
testCases = input(); testCases = (int (testCases))

while(testCases > 0):
    testCases -= 1
    result = []
    #len = input(); len = (int (len))

    inNumber = input()
    inNumber = inNumber.split()

    len = int(inNumber[0])
    addnumber = inNumber[1]

    ogNumber = input()

    flag = 0

    numberAddnumber = int(addnumber)

    for i in range(len):
        if(ogNumber[i] <= addnumber and flag == 0 and numberAddnumber > 1):
            result.append(addnumber)
            flag = 1

        #en el caso de que sea cero
        if ogNumber[i] == "0" and flag == 0 and numberAddnumber > 1 : 
            result.append(addnumber)
            flag = 1

        result.append(ogNumber[i])

    #if the number to be added is zero or one
    numberAddnumber = int(addnumber)
    if(numberAddnumber <= 1 and flag == 0):
        result.append(addnumber)

    result = ''.join(result)
    print(result)